nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -75,6 +78,45 @@ public static DocumentModel moveContentToColdStorage(CoreSession session, Docume
         return session.saveDocument(documentModel);
     }
 
+    /**
+     * Retrieves the cold storage content associated with the document of the given {@link DocumentRef}.
+     * 
+     * @param session the core session
+     * @param documentRef the document reference
+     * @param numberOfDaysOfAvailability number of days that you want your cold storage content to be accessible after
+     *            restoring
+     * @apiNote This method will initiate a restore request, calling the {@link Blob#getStream()} during this process
+     *          doesn't mean you will get the blob's content.
+     * @return the updated document model if the retrieve succeeds
+     * @throws NuxeoException if there is no cold storage content associated with the given document, or if it is being
+     *             retrieved
+     */
+    // TODO currently this method sets only the "coldstorage:beingRetrieved" property, once the NXP-28417 is done,
+    // it will trigger the restore request too
+    public static DocumentModel retrieveContentFromColdStorage(CoreSession session, DocumentRef documentRef,
+            int numberOfDaysOfAvailability) {
+        DocumentModel documentModel = session.getDocument(documentRef);
+        log.debug("Retrieve from cold storage the content of document: {} for: {} days", documentModel,
+                numberOfDaysOfAvailability);
+
+        if (!documentModel.hasFacet(FacetNames.COLD_STORAGE)
+                || documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY) == null) {
+            throw new NuxeoException(String.format("No cold storage content defined for document: %s.", documentModel),
+                    SC_NOT_FOUND);
+        }
+
+        Serializable beingRetrieved = documentModel.getPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY);
+        if (Boolean.TRUE.equals(beingRetrieved)) {
+            throw new NuxeoException(
+                    String.format("The cold storage content associated with the document: %s is being retrieved.",
+                            documentModel),
+                    SC_CONFLICT);
```

Comment:
Not sure about this HTTP status code, as this been discussed? I would see instead 403 or 404 or 412.

AWS itself returns

		HTTP/1.1 403 Forbidden
		<Error>
		   <Code>InvalidObjectState</Code>
		   <Message>The operation is not valid for the object's storage class</Message>
		</Error>